### PR TITLE
task-NXP-25271-RenameFile

### PR DESCRIPTION
**Pull request by ataillefer**

### CODE REVIEW

File: src/main/java/org/nuxeo/wopi/rest/FilesResource.java [hunk 1]
```diff
@@ -273,6 +276,66 @@ protected Object unlockOrRefresh(String fileId, String lock, boolean unlock) {
...
+            Blob blob = getMainBlob(doc);
+            if (blob == null) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+
+            String nameJSON = "{ \"Name\": \"" + requestedName + "\" }";
+            if (!doc.isLocked()) {
+                if (!session.hasPermission(doc.getRef(), SecurityConstants.WRITE_PROPERTIES)) {
+                    // cannot rename blob
+                    return Response.status(Response.Status.CONFLICT).build();
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/wopi/rest/FilesResource.java]: Is it expected to return a conflict instead a forbidden ?

File: src/main/java/org/nuxeo/wopi/rest/FilesResource.java [hunk 2]
```diff
@@ -273,6 +276,66 @@ protected Object unlockOrRefresh(String fileId, String lock, boolean unlock) {
...
+                }
+                blob.setFilename(requestedName);
+                BlobHolder bh = doc.getAdapter(BlobHolder.class);
+                bh.setBlob(blob);
+                doc.putContextData("source", "wopi");
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/wopi/rest/FilesResource.java]: Can you use `CoreSession#SOURCE` ?

File: src/main/java/org/nuxeo/wopi/rest/FilesResource.java [hunk 3]
```diff
@@ -273,6 +276,66 @@ protected Object unlockOrRefresh(String fileId, String lock, boolean unlock) {
...
+                doc = session.saveDocument(doc);
+                return Response.ok(nameJSON).build();
+            }
+
+            KeyValueService service = Framework.getService(KeyValueService.class);
+            KeyValueStore store = service.getKeyValueStore("wopi-locks");
+            String currentLock = store.getString(doc.getId());
+            if (currentLock == null) {
+                // locked by Nuxeo
+                return Response.status(Response.Status.CONFLICT).build();
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/wopi/rest/FilesResource.java]: Is it possible to leverage `X-WOPI-LockFailureReason` for kind of lock ?

Comment (kevinleturc) [reply src/main/java/org/nuxeo/wopi/rest/FilesResource.java]: Also, we might add the `X-WOPI-Lock` ?
>>>409 Conflict – Lock mismatch/locked by another interface; an X-WOPI-Lock response header containing the value of the current lock on the file must always be included when using this response code

File: src/main/java/org/nuxeo/wopi/rest/FilesResource.java (hunk 1 above)
Comment (troger) [reply src/main/java/org/nuxeo/wopi/rest/FilesResource.java]: WOPI does not expect forbidden.

File: src/main/java/org/nuxeo/wopi/rest/FilesResource.java (hunk 3 above)
Comment (troger) [reply src/main/java/org/nuxeo/wopi/rest/FilesResource.java]: When the document is locked by Nuxeo, so not by WOPI, there is no lock value.
The `X-WOPI-LockFailureReason` is optional and it's only for logging purposes, and we don't even know if OO uses it. The point here is to have OO prompts the user that he can't rename the file.

**Comment by troger** [comment]
:+1:

**Pull request merged**